### DISCUSSION TITLE
remove Jetbrain Cache

### DISCUSSION THREAD

**Discussion by Pasanlaksitha** [senurah/java-tess4J-ocr#2] [Ideas]
I suggest below.git ignore template for your project. This will be safe and clean to contribute 

target/
!.mvn/wrapper/maven-wrapper.jar
!**/src/main/**/target/
!**/src/test/**/target/

### IntelliJ IDEA ###
.idea/modules.xml
.idea/jarRepositories.xml
.idea/compiler.xml
.idea/libraries/
*.iws
*.iml
*.ipr

### Eclipse ###
.apt_generated
.classpath
.factorypath
.project
.settings
.springBeans
.sts4-cache

### NetBeans ###
/nbproject/private/
/nbbuild/
/dist/
/nbdist/
/.nb-gradle/
build/
!**/src/main/**/build/
!**/src/test/**/build/

### VS Code ###
.vscode/

### Mac OS ###
.DS_Store

### Jetbrains ### 
.idea

**Comment by senurah** [comment]
Can you specify the reason?

**Reply by Pasanlaksitha** [reply]
Removing the `.idea` folder and adding it to `.gitignore` keeps the repository clean by avoiding IDE-specific settings from being shared. It prevents unnecessary merge conflicts, maintains cross-IDE compatibility, and ensures version control focuses on actual project files, not local environment configurations.

**Reply by senurah** [reply]
Ok, I'll check on that

**Comment by Pasanlaksitha** [comment]
By removing the `.idea` folder from the repository and adding it to `.gitignore`, you:

- Avoid sharing personal IDE settings** that aren't useful for others.
- Prevent merge conflicts** from constant updates to IDE files.
- Maintain flexibility** for teammates using different editors (VSCode, Sublime, etc.).
- Keep the repo size smaller** and focus only on project-relevant files.

It's a best practice to keep the repository clean and collaborative without forcing others to deal with irrelevant local configurations.
